patches/api/0449-add-number-format-api.patch

```diff
@@ -0,0 +1,194 @@
++
++    StyledFormat PLAYER_LIST_DEFAULT = NumberFormat.styled(Style.empty().color(NamedTextColor.YELLOW));
++
++    /*
++     * The style that is being applied to the number in the score
++     * @return the style to apply
++     */
++    Style getStyle();
++
++}
+diff --git a/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormatImpl.java b/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormatImpl.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..e25c43608136c4f46a5f968d24a13916fb240510
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormatImpl.java
+@@ -0,0 +1,11 @@
++package io.papermc.paper.scoreboard.numbers;
++
++import net.kyori.adventure.text.format.Style;
++
++record StyledFormatImpl(Style style) implements StyledFormat {
++
++    @Override
++    public Style getStyle() {
++        return style;
++    }
++}
+diff --git a/src/main/java/org/bukkit/scoreboard/Objective.java b/src/main/java/org/bukkit/scoreboard/Objective.java
+index a193ffabb05160b462dee1ba8f687fdbc84405b6..ad2fe248ebbd37b22a04a12a509d4399f62daf2e 100644
+--- a/src/main/java/org/bukkit/scoreboard/Objective.java
++++ b/src/main/java/org/bukkit/scoreboard/Objective.java
+@@ -175,4 +175,18 @@ public interface Objective {
+      */
+     @NotNull Score getScoreFor(@NotNull org.bukkit.entity.Entity entity) throws IllegalArgumentException, IllegalStateException;
+     // Paper end - improve scoreboard entries
++
++    //Paper start - number format
++    /*
++     * Gets the number format for this objective's entries (scores)
++     * @return this objective's number format
++     */
++    @Nullable io.papermc.paper.scoreboard.numbers.NumberFormat getNumberFormat();
++
++    /*
++     * Sets the number format for this objective's entries (scores)
++     * @param format the number format to set, pass null to reset format to default
++     */
++    void setNumberFormat(@Nullable io.papermc.paper.scoreboard.numbers.NumberFormat format);
++    //Paper end
+ }
+diff --git a/src/main/java/org/bukkit/scoreboard/Score.java b/src/main/java/org/bukkit/scoreboard/Score.java
+index 1eaa9a93f8eff5f18a6cce2d74f21eb19db273c8..86f70362a8852d06a378b5dd827818d330f2a496 100644
+--- a/src/main/java/org/bukkit/scoreboard/Score.java
++++ b/src/main/java/org/bukkit/scoreboard/Score.java
+@@ -83,4 +83,19 @@ public interface Score {
+      */
+     void resetScore() throws IllegalStateException;
+     // Paper end
++
++    //Paper start - number format
++    /*
++     * Gets the number format for this score
++     * @return this score's number format
++     */
++    @Nullable io.papermc.paper.scoreboard.numbers.NumberFormat getNumberFormat();
++
++    /*
++     * Sets the number format for this score
++     * @param format the number format to set, pass null to reset format to default
++     */
++    void setNumberFormat(@Nullable io.papermc.paper.scoreboard.numbers.NumberFormat format);
++    //Paper end
```

Comment:
Each of these should have `@throws IllegalStateException if this objective has been unregsitered`.